src/recipe.jl

```diff
@@ -110,6 +113,53 @@ function Makie.plot!(plot::Beeswarm)
         calculate!(point_buffer.val, algorithm, direction == :y ? pixelspace_point_buffer.val : reverse.(pixelspace_point_buffer.val), markersize, side)
         # Project the beeswarm back to data space and store it, again, in `point_buffer.val`
         point_buffer.val .= Point2f.(Makie.project.((scene.camera,), :pixel, :data, direction == :y ? (point_buffer.val) : reverse.(point_buffer.val)))
+
+        # Method to create a gutter when a gutter is defined
+        # NOTE: Maybe turn this into a helper function?
+
+        if !isnothing(plot.attributes[:gutter].val)
+            # Get category labels
+            groups = [pt[1] for pt in positions] 
+
+            # Sort positions before iterating through values
+            sort!([positions], by = x -> x[1])
+
+            # Find all points belonging to all unique categories
+            idx = 1
+            for group in unique(groups) |> sort
+                
+                # Starting index for the group
+                starting = findfirst(==(group), groups)
+
+                # Last index for the group
+                ending = findlast(==(group), groups)
```

Comment:
It's not guaranteed that groups are continuous (see e.g. the iris example) so I would recommend either `findall(==(group), xs)` or `xs .== group` as a boolean mask.

Reply:
Great point -- that's why I had sorting but this is so much cleaner. Can't believe I didn't think of this method. :joy: